flang/lib/Lower/OpenMP/ClauseProcessor.cpp

```diff
@@ -620,7 +620,7 @@ class TypeInfo {
   }
 
   // Returns the shape of array types.
-  const llvm::SmallVector<int64_t> &getShape() const { return shape; }
+  llvm::ArrayRef<int64_t> getShape() const { return shape; }
```

Comment:
You don't really gain much by returning `ArrayRef` for a `const Container &`.  The flexibility of ArrayRef comes from the ability to convert multiple containers into it, the reverse isn't true (there is only a conversion to `std::vector` available).

I'll leave it up to you what you want to do with it.

Reply:
I get what you say. The idea with this change was mainly to avoid giving users too much detail on how the shape is stored internally. They shouldn't care whether it is stored as a `SmallVector`, an `std::vector` or anything else. By returning a more generic `ArrayRef` interface we prevent users from relying on `SmallVector`-specific features, allowing us to easily change the container if we need to later on. So, if that's ok, I'd prefer keeping the change.

Reply:
Sure, I'm ok with whatever you decide.